jose-jwt/jwa/EcdhKeyManagementWin.cs

```diff
@@ -1,88 +1,99 @@
+        public EcdhKeyManagementWin(bool isDirectAgreement, EcdhKeyManagementUnix ecdhKeyManagementUnix)
         {
+            this.ecdhKeyManagementUnix = ecdhKeyManagementUnix;
             algIdHeader = isDirectAgreement ? "enc" : "alg";
         }
 
         public virtual byte[][] WrapNewKey(int cekSizeBits, object key, IDictionary<string, object> header)
         {
+            if (key is ECDiffieHellman)
+            {
+                return ecdhKeyManagementUnix.WrapNewKey(cekSizeBits, key, header);
+            }
+            
             var cek = NewKey(cekSizeBits, key, header);
             var encryptedCek = Wrap(cek, key);
 
-            return new[] { cek, encryptedCek };
+            return new[] { cek, encryptedCek };
         }
 
         public virtual byte[] WrapKey(byte[] cek, object key, IDictionary<string, object> header)
-        {
-            throw new JoseException("(Direct) ECDH-ES key management cannot use existing CEK.");
-        }
-
+        {
+            throw new JoseException("(Direct) ECDH-ES key management cannot use existing CEK.");
+        }
+
         private byte[] NewKey(int keyLength, object key, IDictionary<string, object> header)
         {
             CngKey recieverPubKey = null;
 
-            if (key is Jwk jwk)
-            {
-                if (jwk.Kty == Jwk.KeyTypes.EC)
-                {
-                    recieverPubKey = jwk.CngKey(CngKeyUsages.KeyAgreement);
-                }
+            if (key is Jwk jwk)
+            {
+                if (jwk.Kty == Jwk.KeyTypes.EC)
+                {
+                    recieverPubKey = jwk.CngKey(CngKeyUsages.KeyAgreement);
+                }
             }
 
             recieverPubKey = recieverPubKey ?? Ensure.Type<CngKey>(key, "EcdhKeyManagement alg expects key to be of CngKey or Jwk types with kty='EC'.");
             
             EccKey ephemeral = EccKey.Generate(recieverPubKey);
 
-            IDictionary<string, object> epk = new Dictionary<string, object>();
+            IDictionary<string, object> epk = new Dictionary<string, object>();
             epk["kty"] = "EC";
             epk["x"] = Base64Url.Encode(ephemeral.X);
             epk["y"] = Base64Url.Encode(ephemeral.Y);
             epk["crv"] = ephemeral.Curve();
 
-            header["epk"] = epk;
+            header["epk"] = epk;
 
-            return DeriveKey(header, keyLength, recieverPubKey, ephemeral.Key);
+            return DeriveKey(header, keyLength, recieverPubKey, ephemeral.Key);
         }
 
         public virtual byte[] Wrap(byte[] cek, object key)
         {
             return Arrays.Empty;
         }
 
-        public virtual byte[] Unwrap(byte[] encryptedCek, object key, int cekSizeBits, IDictionary<string, object> header)
+        public virtual byte[] Unwrap(byte[] encryptedCek, object key, int cekSizeBits, IDictionary<string, object> header)
         {
+            if (key is ECDiffieHellman)
```

Comment:
is this to support new `ECDiffieHellman` type of key on Windows in addition to `CngKey` and `Jwk` ?

Reply:
Exactly.